### DISCUSSION TITLE
Creating new background service during runtime

### DISCUSSION THREAD

**Discussion by jasonycw** [dotnet/aspnetcore#26325] [General]
I know that there is `IHostedService` that can be used for implementing background tasks by adding it in Startup with `.AddHostedService`.

However, there seems to be very little about how to create a background service on demand during runtime, after startup.

For example, after an MVC API application is started up, when it receives a specific request, I want to have a background service to be created for that request to do some background tasks and not blocking the the controller thread.
That way, the API will return something like `202 Accepted` immediately and then the background job will handle the request.

Is there such feature? And are there any documentation about it?

FYI, I am using .NET Core 3.1

**Comment by Tratcher** [comment]
They this:
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.1&tabs=visual-studio#queued-background-tasks

**Reply by jasonycw** [reply]
Yes, I did come across this document, but it's more like **one** background service to handle the lambda expression in the `ConcurrentQueue<Func<CancellationToken, Task>>` under `IBackgroundTaskQueue`. 
We did try to implement something similar while trying out difference solution but in our use case, this doesn't feels suitable.

We are trying to create a new background service on the fly instead of queuing up `Task`, since that background service is suppose to do something complicated and require proper DI, We actually like to have proper life cycle for those background services even they are not initialized on startup.

**Reply by Tratcher** [reply]
There's no built in feature for that. I suggest starting with the example above and expanding on it. You can inject the IServiceProvider into the constructor and manage DI for your later tasks from there.

**Reply by davidfowl** [reply]
> We did try to implement something similar while trying out difference solution but in our use case, this doesn't feels suitable.

Can you elaborate? What doesn't feel suitable? What was missing?
